src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -324,7 +324,7 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
               val idxOpt = intConstantEval(index)
               error(n, s"index $index is out of bounds", !idxOpt.forall(i => i >= 0 && i < l))
 
-            case (PointerT(ArrayT(l, _)), IntT(_)) =>
+            case (ActualPointerT(ArrayT(l, _)), IntT(_)) =>
```

Comment:
I think a ghost pointer to an array is ok, i.e. this line should account for ghost pointer too.